File: code_samples/python/v1.0.0/loan_precalculation.py
from tatrapayplus.client import TBPlusSDK
from tatrapayplus.models import *

tatrapay_client = TBPlusSDK(
    client_id="your-client-id",
    client_secret="your-client-secret",
    mode=Mode.PRODUCTION,
)

loan_data = BasicCalculationRequest(
    loan_amount=250.45,
    payment_method=BasicCalculationRequestPaymentMethod.PAY_LATER,
    capacity_info=CapacityInfo(  # not required
        monthly_income=2000.0,
        monthly_expenses=800.0,
        number_of_children=1,
    )
)

loan_offers = tatrapay_client.precalculate_loan(
    request=loan_data,
    ip_address="127.0.0.1",
)

for loan_offer in loan_offers:
    # loan_offer = BasicCalculationResponseItem object
    print(loan_offer.loan_duration)
    print(loan_offer.loan_interest_rate)
    # ...

File: code_samples/php/v2/loan_precalculation.php
<?php
use Tatrapayplus\TatrapayplusApiClient\Api\TatraPayPlusAPIApi;
use Tatrapayplus\TatrapayplusApiClient\Model\PaymentMethod;
use Tatrapayplus\TatrapayplusApiClient\Model\BasicCalculationRequest;
use Tatrapayplus\TatrapayplusApiClient\Model\CapacityInfo;

$tatrapayplus_api = new TatraPayPlusAPIApi(
    "your-client-id",
    "your-client-secret",
    mode: TatraPayPlusAPIApi::PRODUCTION
);
$request = new BasicCalculationRequest([
    "payment_method" => PaymentMethod::PAY_LATER,
    "loan_amount" => 150.45,
    "capacity_info" => new CapacityInfo([ // not required
        "monthly_income" => 0,
        "monthly_expenses" => 0,
        "number_of_children" => 0
    ])
]);

$response = $tatrapayplus_api->loanPrecalculation($request);

$response["response"]->getStatusCode(); // 200 = success

$response_list = $response['object']; // contains a list of BasicCalculationResponseItem objects
foreach ($response_list as $item) {
    $item->getLoanInterestRate();
    $item->getLoanDuration();
    // ....
}